### DISCUSSION TITLE
Improving web vitals

### DISCUSSION THREAD

**Discussion by vedranmiletic** [squidfunk/mkdocs-material#6764] [Ideas]
Google's [PageSpeed Insights](https://pagespeed.web.dev/analysis/https-squidfunk-github-io-mkdocs-material/vg7ces5pju?form_factor=mobile) gives Material for MkDocs a good, but not great score. Could it be improved?

**Comment by squidfunk** [comment]
Thanks for asking. It is always good to measure web vitals, but I'd say that our landing page might not be a representative candidate, because it is a heavily customized template with a lot of images. If you check, for example, the [getting started guide](https://squidfunk.github.io/mkdocs-material/getting-started/), things should be more representative.

Also note that Google Fonts on mobile does take some time on the first load, and since we're swapping the font upon load, it counts towards the largest contentful paint after 2.7s. This makes the entire statistics look rather bad, albeit it has no impact on the UI/UX. That's at least my interpretation. PageSpeed and Lighthouse should always be taken with a grain of salt.

<img width="572" alt="Bildschirm­foto 2024-02-12 um 09 03 01" src="https://github.com/squidfunk/mkdocs-material/assets/932156/67319f20-640b-42a2-a837-422765061b05">

**Reply by alexvoss** [reply]
I would add that any optimization may come with drawbacks, too and the costs vs. benefits would need to be weighed. We had an interesting discussion recently (can't find it now, typical...) about the size of the DOM, which is affected by the entire navigation being included in pages so that mobile navigation can work. Changing that might harm the very people we are looking to help here. 

Another aspect is the loading of the search index. It allows for pretty much instant search results, which is an awesome thing to have. Technical measures are nice as they are easy to do thanks to Google but in the end we need to ask user-centric questions. 

Just my 2p.

**Reply by squidfunk** [reply]
Maintaining this project for 8 years now, my learning is: if you optimize one thing, somebody else will come and want something else optimized 😅 We always have to strike a balance.

**Comment by vedranmiletic** [comment]
Fair points, thanks for taking time to reply.
